Source/qol/stash.cpp

```diff
@@ -352,21 +352,19 @@ void DrawStash(const Surface &out)
 
 	constexpr Displacement offset { 0, INV_SLOT_SIZE_PX - 1 };
 
-	for (auto slot : StashGridRange) {
-		StashStruct::StashCell itemId = Stash.GetItemIdAtPosition(slot);
-		Item &item = Stash.stashList[itemId];
-		if (Stash.IsItemAtPosition(slot)) {
-			InvDrawSlotBack(out, GetStashSlotCoord(slot) + offset, InventorySlotSizeInPixels, item._iMagical);
-		}
-	}
-
 	for (auto slot : StashGridRange) {
 		StashStruct::StashCell itemId = Stash.GetItemIdAtPosition(slot);
 		if (itemId == StashStruct::EmptyCell) {
 			continue; // No item in the given slot
 		}
 
 		Item &item = Stash.stashList[itemId];
+
+		// provide item background coloring
+		if (Stash.IsItemAtPosition(slot)) {
+			InvDrawSlotBack(out, GetStashSlotCoord(slot) + offset, InventorySlotSizeInPixels, item._iMagical);
+		}

```

Comment:
https://github.com/diasurgical/devilutionX/blob/a2a8b1c706789a9f7ddc08c7e4941d69d43690aa/Source/qol/stash.h#L50-L53

This loop is already doing this same validation right at the start.

```suggestion
		InvDrawSlotBack(out, GetStashSlotCoord(slot) + offset, InventorySlotSizeInPixels, item._iMagical);
```